client/src/util/exponentialBackoffFetch.js

```diff
@@ -0,0 +1,68 @@
+export default function createBackoffFetch(retryTimes = 3) {
+  let prevHash = null;
+  let retryCount = 0;
+  let response = null;
+  let error = null;
+  let promise = null;
+
+  // This will only be called by itself or when we've seen a new matrix
+  const backoffFetch = (fetchArgs) => {
+    const promiseExecuter = (resolve, reject) => {
+      retryCount += 1;
+      // We don't want to timeout our first request
+      if (retryCount === 1) {
+        fetch(...fetchArgs)
+          .then((resp) => {
+            response = resp;
+            if (resp.ok) {
+              resolve(response);
+            } else {
+              promiseExecuter(resolve, reject);
+            }
+          })
+          .catch((e) => {
+            response = null;
+            error = e;
+            promiseExecuter(resolve, reject);
+          });
+      } else {
+        // Timeout all other requests
+        setTimeout(async () => {
+          try {
+            const resp = await fetch(...fetchArgs);
+            response = resp;
+            // Fetch status 2xx
+            if (resp.ok) {
+              resolve(response);
+            } else if (retryCount - 1 === retryTimes) {
+              if (error) reject(error);
+              resolve(response);
+            } else {
+              // Some non 2xx response code
+              promiseExecuter(resolve, reject);
+            }
+          } catch (e) {
+            // Fetch API error
+            error = e;
+            if (retryCount - 1 === retryTimes) reject(error);
+            promiseExecuter(resolve, reject);
+          }
+          // delay amount (3.5, 17.5, 87.5, 437.5 seconds)
```

Comment:
if we default to retry 3 times, then is it just `0, 3.5, 17.5, and 87.5`?

Reply:
Yep! I forget why I originally gave four examples, but I left the last one in since the retry amount is tweakable

Reply:
That makes sense! Thanks!